programs/x509/cert_write.c

```diff
@@ -175,7 +191,9 @@ struct options
     int basic_constraints;      /* add basic constraints ext to CRT     */
     int version;                /* CRT version                          */
     mbedtls_md_type_t md;       /* Hash used for signing                */
+    mbedtls_md_type_t sig_alg;  /* MD to use generating signature       */
```

Comment:
```suggestion
    mbedtls_md_type_t sig_hash;  /* MD to use generating signature       */
```
`sig_alg` is used by TLS1.3, that not only includes hash algorithm. This field is different with `sig_alg` in TLS1.3.

Reply:
The X.509 code uses `sig_alg` and not `sig_hash`, so this is consistent with the terminology in the X.509 library code. So I think this is right.

Reply:
ping @yuhaoth ?

Reply:
```
$ fgrep md_alg library/x509*.c | wc
      35     256    2835
$ fgrep sig_alg library/x509*.c | wc
      12     104    1132
$ fgrep sig_hash library/x509*.c | wc
       0       0       0
```

`sig_alg` is fine. `md_alg` might be better. but I am fine with leaving this as-is

Reply:
`sig_alg` is the choice of PK algorithm (RSA vs ECC). It doesn't make sense as a variable name for a value of type `mbedtls_md_type_t`, which only indicates a choice of _hash_ algorithm.